lib/pbench/server/cache_manager.py

```diff
@@ -882,6 +882,16 @@ def create(self, controller_name: str, tarfile: Path) -> Tarball:
         Returns
             Tarball object
         """
+        try:
+            metadata = Tarball._get_metadata(tarfile)
+            controller_name = metadata["run"]["controller"]
```

Comment:
I think a `KeyError` here should be caught explicitly rather than falling to the `except Exception`: `extractfile` can raise a `KeyError` because `{name}/metadata.log` doesn't exist (which should really be handled in `Tarball.extract`), *and* we don't know that `metadata` has a `"run"` or that the `run` dict has a `"controller"`. `_get_metadata` should also recognize that we might get `tarfile.TarError` exceptions (it's not really detailed, although it suggests that it usually just returns `None`).